exasol_advanced_analytics_framework/query_handler/context/proxy/bucketfs_location_proxy.py

```diff
@@ -22,20 +21,20 @@ def cleanup(self):
             raise Exception("Cleanup of BucketFSLocationProxy only allowed after release.")
         files = self._list_files()
         for file in files:
+            print(f'BucketFSLocationProxy.cleanup() remove {file}')
             self._remove_file(file)
 
     def _remove_file(self, file):
         try:
-            self._bucketfs_location.delete_file_in_bucketfs(file)
+            file.rm()
         except Exception as e:
             LOGGER.error(f"Failed to remove {file}, got exception", exc_info=True)
 
     def _list_files(self):
-        files = []
         try:
-            files = self._bucketfs_location.list_files_in_bucketfs("")
+            return list(self._bucketfs_location.iterdir())
```

Comment:
This will also include subdirectories I think. Is it OK?

Reply:
Excellent question. :+1: 

AFAIK, the old implementation had the same semantics, see the [related test](https://github.com/exasol/bucketfs-python/blob/main/tests/unit_tests/test_localfs_mock_bucketfs_location.py#L235).